src/main/client_context.cpp

```diff
@@ -532,11 +532,16 @@ unique_ptr<PendingQueryResult> ClientContext::PendingPreparedStatement(ClientCon
 	return PendingPreparedStatementInternal(lock, prepared, parameters);
 }
 
+void ClientContext::WaitForTask(ClientContextLock &lock, BaseQueryResult &result) {
+	active_query->executor->WaitForTask();
+}
+
 PendingExecutionResult ClientContext::ExecuteTaskInternal(ClientContextLock &lock, BaseQueryResult &result,
                                                           bool dry_run) {
 	D_ASSERT(active_query);
 	D_ASSERT(active_query->IsOpenResult(result));
 	bool invalidate_transaction = true;
+	WaitForTask(lock, result);
```

Comment:
Given that we are also using this to update the progress bar - maybe we can have a timeout in the `wait` (i.e. using `wait_until` with e.g. a `20ms` timeout)?

Reply:
That makes sense 👍